wb/main/jobs/datasets/recognize_dataset_job.py

```diff
@@ -42,16 +46,42 @@ def run(self):
                 return
             # Early exit if dataset type is already defined by user
             if not dataset.dataset_type:
-                dataset_path = dataset.path
-                dataset.dataset_type = self._recognize(dataset_path)
+                dataset.dataset_type = self._recognize(dataset)
             dataset.write_record(session)
             self._job_state_subject.update_state(status=StatusEnum.ready, progress=100)
             self._job_state_subject.detach_all_observers()
 
-    @staticmethod
-    def _recognize(path: str):
-        for dataset_type, dataset_recognizer in ConfigRegistry.dataset_recognizer_registry.items():
-            if dataset_recognizer.recognize(path):
-                return dataset_type
-        remove_dir(path)
-        raise UnknownDatasetError('Unknown dataset type')
+    def _recognize(self, dataset: DatasetsModel) -> DatasetTypesEnum:
+        if not DATASET_REPORTS_FOLDER.exists():
+            create_empty_dir(DATASET_REPORTS_FOLDER)
+        report_path = DATASET_REPORTS_FOLDER / f'{dataset.id}.json'
+
+        tool = DatumaroTool()
+        tool.set_mode(DatumaroModesEnum.detect_format)
+        tool.set_path('json-report', report_path)
```

Comment:
again - create a new mwthod

Reply:
We'll be getting a lot more parameters later down the line for things such as TFDS info and downloading. Both are restricted to either a flag or a path argument. Wouldn't this cause a massive function bloat inside the tool class? These generic functions were designed with this in mind.